File: Support/Sensor/sensorsdataUnityBridge.java
package com.sensorsdata.Unity;


import org.json.JSONException;
import org.json.JSONObject;

import android.app.Activity;

import com.sensorsdata.analytics.android.sdk.SensorsDataAPI;
import com.sensorsdata.analytics.android.sdk.exceptions.InvalidDataException;

/**
 * Created by 张昊阳 on 2016/12/23
 */

public class sensorsdataUnityBridge 
{
	private static Activity m_UnityActive;
	
    public static void initSensorsDataAPI(Activity Context , String serverurl, String configurl, int isDebug) 
    {
    	SensorsDataAPI.DebugMode debugmode = SensorsDataAPI.DebugMode.DEBUG_OFF;
		if(isDebug == 1)
		{
			debugmode = SensorsDataAPI.DebugMode.DEBUG_AND_TRACK;
		}
		m_UnityActive = Context;
		SensorsDataAPI.sharedInstance(Context,serverurl,configurl,debugmode);
		SensorsDataAPI.sharedInstance(m_UnityActive).enableAutoTrack();
    }
	    
    public static void TrackEvent(String EventName, String EventJson)
    {
	   try 
	   {
	       JSONObject Json = new JSONObject(EventJson);
	       SensorsDataAPI.sharedInstance(m_UnityActive).track(EventName, Json);
	   }
	   catch (InvalidDataException e) 
	   {
	       e.printStackTrace();
	   }
	   catch (JSONException e) 
	   {
		   e.printStackTrace();
	   }
    }
    
    public static void TrackEvent(String EventName)
    {
	   try 
	   {
	       JSONObject Json = new JSONObject();
	       SensorsDataAPI.sharedInstance(m_UnityActive).track(EventName, Json);
	   }
	   catch (InvalidDataException e) 
	   {
	       e.printStackTrace();
	   }
    }
    
    public static void SetUserId(String userid) 
    {
    	try
    	{
    		SensorsDataAPI.sharedInstance(m_UnityActive).login(userid);
    	}
    	catch(InvalidDataException e)
    	{
    		e.printStackTrace();
    	}
	}
    
    public static void SetGameInfo(String channel, String codename)
    {
	    try 
	    {
	    	JSONObject properties = new JSONObject();
	    	properties.put("Channel", channel);
	    	properties.put("codename", codename);
	    	SensorsDataAPI.sharedInstance(m_UnityActive).profileSetOnce(properties);	
	    }
	    catch (InvalidDataException e) 
	    {
	    	e.printStackTrace();
	    }
	    catch (JSONException e) 
	    {
	    	e.printStackTrace();
	    }
    }
    
    public static void SetProfile(String name, String value) 
    {
    	try 
 	    {
 	    	JSONObject properties = new JSONObject();
 	    	properties.put(name, value);
 	    	SensorsDataAPI.sharedInstance(m_UnityActive).profileSet(properties);	
 	    }
 	    catch (InvalidDataException e) 
 	    {
 	    	e.printStackTrace();
 	    }
 	    catch (JSONException e) 
 	    {
 	    	e.printStackTrace();
 	    }
	}
    
    public static void Logout()
    {
    	SensorsDataAPI.sharedInstance(m_UnityActive).logout();
    }
    
    public static void TrackInstallation()
    {
	   try 
	   {
	      JSONObject properties = new JSONObject();

	      // 追踪渠道效果
	      SensorsDataAPI.sharedInstance(m_UnityActive).trackInstallation("AppInstall", properties);
	    } catch (Exception e) {
	      e.printStackTrace();
	    }
    }
}

File: Assets/Plugins/iOS/SensorsData/SensorsDataUnityBridge.m
//
//  SensorsDataUnityBridge.m
//  SensorsAnalyticsSDK
//
//  Created by zhanghaoyang on 16/12/26.
//  Copyright © 2016年 SensorsData. All rights reserved.
//

#import "SensorsDataUnityBridge.h"

#if defined(__cplusplus)
extern "C"{
#endif
    void _InitSensorsData(char serverurl[], char configurl[], BOOL isdebug)
    {
        NSString *ServerUrl = [NSString stringWithUTF8String:serverurl];
        NSString *ConfigUrl = [NSString stringWithUTF8String:configurl];
        [SensorsDataUnityBridge InitSensorsData:ServerUrl sec:ConfigUrl thr:isdebug];
    }
    
    void _TrackEvent(char eventname[], char jsonstring[])
    {
        NSString *EventName = [NSString stringWithUTF8String:eventname];
        NSString *JsonString = [NSString stringWithUTF8String:jsonstring];
        [SensorsDataUnityBridge TrackEvent:EventName sec:JsonString];
    }
    
//    void _TrackEventName(char eventname[])
//    {
//        NSString *EventName = [NSString stringWithUTF8String:eventname];
//        [SensorsDataUnityBridge TrackEvent:EventName];
//    }
    
    void _SetUserId(char userid[])
    {
        NSString *Userid = [NSString stringWithUTF8String:userid];
        [SensorsDataUnityBridge SetUserId:Userid];
    }
    
    void _SetGameInfo(char channel[], char code[])
    {
        NSString *Channel = [NSString stringWithUTF8String:channel];
        NSString *Code = [NSString stringWithUTF8String:code];
        [SensorsDataUnityBridge SetGameInfo:Channel sec:Code];
    }

    void _SetProfile(char name[], char value[])
    {
        NSString *Name = [NSString stringWithUTF8String:name];
        NSString *Value = [NSString stringWithUTF8String:value];
        [SensorsDataUnityBridge SetProfile:Name va:Value];
    }

    void _SetDicProfile(char jsonstring[])
    {
        NSString *JsonString = [NSString stringWithUTF8String:jsonstring];
        [SensorsDataUnityBridge SetDicProfile:JsonString];
    }

    void _TrackInstallation()
    {
        [SensorsDataUnityBridge TrackInstallation];
    }
#if defined(__cplusplus)
}
#endif

@implementation SensorsDataUnityBridge : NSObject

static SensorsDataUnityBridge * instance;
+(instancetype)INSTANCE{
    if(instance == nil){
        instance = [[SensorsDataUnityBridge alloc] init];
    }
    return instance;
}

+(void)InitSensorsData:(NSString *)SERVERURL sec:(NSString *)CONFIGURL thr:(BOOL)isdebug
{
    // Debug 模式选项
    //   SensorsAnalyticsDebugOff - 关闭 Debug 模式
    //   SensorsAnalyticsDebugOnly - 打开 Debug 模式，校验数据，但不进行数据导入
    //   SensorsAnalyticsDebugAndTrack - 打开 Debug 模式，校验数据，并将数据导入到 Sensors Analytics 中
    // 注意！请不要在正式发布的 App 中使用 Debug 模式！
    SensorsAnalyticsDebugMode debugmode = SensorsAnalyticsDebugOff;
    if (isdebug)
    {
        debugmode = SensorsAnalyticsDebugAndTrack;
    }
    [SensorsAnalyticsSDK sharedInstanceWithServerURL:SERVERURL
                                     andConfigureURL:CONFIGURL
                                        andDebugMode:debugmode];
    
    [[SensorsAnalyticsSDK sharedInstance] enableAutoTrack];
}

+(void)TrackEvent:(NSString *)EventName sec:(NSString *)JsonString
{
    NSData* jsonData = [JsonString dataUsingEncoding:NSUTF8StringEncoding];
    NSMutableDictionary* dic = [NSJSONSerialization JSONObjectWithData:jsonData options:NSJSONReadingMutableContainers error:nil];
    
    [[SensorsAnalyticsSDK sharedInstance] track:EventName withProperties:dic];
}

//+(void)TrackEvent:(NSString *)EventName
//{
//    [[SensorsAnalyticsSDK sharedInstance] track:EventName withProperties:nil];
//}

+(void)SetUserId:(NSString *)UserId
{
    [[SensorsAnalyticsSDK sharedInstance] login:UserId];
}

+(void)SetGameInfo:(NSString *)channel sec:(NSString *)codename
{
    [[[SensorsAnalyticsSDK sharedInstance] people] setOnce:@"Channel" to:channel];
    [[[SensorsAnalyticsSDK sharedInstance] people] setOnce:@"codename" to:codename];
}

+(void)SetProfile:(NSString *)name va:(NSString *)value
{
    [[[SensorsAnalyticsSDK sharedInstance] people] set:name to:value];
}

+(void)SetDicProfile:(NSString *)JsonString
{
    NSData* jsonData = [JsonString dataUsingEncoding:NSUTF8StringEncoding];
    NSMutableDictionary* dic = [NSJSONSerialization JSONObjectWithData:jsonData options:NSJSONReadingMutableContainers error:nil];
    
    [[[SensorsAnalyticsSDK sharedInstance] people] set:dic];
}

+(void)TrackInstallation
{
    // 获取当前时间
    NSDate *currentTime = [NSDate date];
    // 在 App 首次启动时，追踪 App 激活事件，并记录首次激活事件
    // AppInstall 事件名称，此处仅仅是一个示例，可以根据实际需求自定义事件名称
    // FirstUseTime 属性仅仅是一个示例，可以根据实际需求添加其它相关属性
    [[SensorsAnalyticsSDK sharedInstance] trackInstallation:@"AppInstall" 
                                             withProperties:@{
                                                 @"FirstUseTime" : currentTime}];
}
@end
